src/lib/RTC.ts

```diff
@@ -0,0 +1,249 @@
+        if (e.candidate) {
+            socket.emit(Constants.ICE_CANDIDATE, e.candidate);
+        }
+    }
+
+    /**
+     * Error handling for creating SDP.
+     */
+    handleCreateDescriptionError = (error: Error): void => {
+        console.log(`Unable to create an offer due to error: ${error.toString()}`);
+    };
+
+    /**
+     * Error handling for setting SDP.
+     */
+    handleSetDescriptionError = (error: Error): void => {
+        console.log(`Something went wrong when setting remote description: ${error.toString()}`);
+    }
+
+    /**
+     * Error handling for creating answer for an offer.
+     */
+    handleCreateAnswerError = (error: Error): void => {
+        console.log(`Something went wrong when creating Answer to Offer. ${error.toString()}`);
+    }
+
+    /**
+     * Error handling for adding ICE candidate.
+     */
+    handleAddCandidateError = (error: Error): void => {
+        console.log(`Something went wrong with adding the candidate. ${error.toString()}`);
+    };
+
+    /**
+     * Callback for when receiveChannel is ready to be set.
+     */
+    receiveChannelCallback = (event: RTCDataChannelEvent): void => {
+        this.receiveChannel = event.channel;
+        this.receiveChannel.onopen = this.handleReceiveChannelStatusChange;
+        this.receiveChannel.onclose = this.handleReceiveChannelStatusChange;
+        this.receiveChannel.onmessage = this.onReceiveMessageCallback;
+    };
+
+    /**
+     * Callback for when data is received. Calls custom handler if there is one.
+     */
+    onReceiveMessageCallback = (event: MessageEvent): void => {
+        console.log(`Received message from receive channel: ${event}`);
+        if (this.customReceiveMessageHandler) {
+            this.customReceiveMessageHandler(event);
+        }
+    }
+
+    /**
+     * Sets custom handler for status change on send channel
+     */
+    setHandleSendChannelStatusChange = (handler: any): void => {
+        this.customSendChannelStatusHandler = handler;
+    }
+
+    /**
+     * Sets custom handler for status change on receive channel
+     */
+    setHandleReceiveChannelStatusChange = (handler: any): void => {
+        this.customReceiveChannelStatusHandler = handler;
+    }
+
+    /**
+     * Sets custom handler for on message received.
+     */
+    setReceiveMessageHandler = (handler: any): void => {
+        this.customReceiveMessageHandler = handler;
+    }
+
+    /**
+     * Sets binary type for send channel
+     */
+    setSendChannelBinaryType = (type: string) => {
+        if (this.sendChannel) {
+            this.sendChannel.binaryType = type;
+        }
+    }
+
+    /**
+     * Sets binary type for receive channel
+     */
+    setReceiveChannelBinaryType = (type: string) => {
+        if (this.receiveChannel) {
+            this.receiveChannel.binaryType = type;
+        }
+    }
+}
+
+const RTCInstance = new RTC();
+export default RTCInstance;
```

Comment:
Wouldn't it be better to make the export the class itself, and make each component create the RTCInstance? That way, we can have multiple rtc instances which addresses that TODO you have in the PR description.

Reply:
Yea, I was coding on the assumption that there would only be one connection (my bad).